### PR TITLE
Add optional namespace prefix for generated modules

### PR DESCRIPTION
**Pull request by johnfoconnor**
the optional `proto_namespace` parameter will be prepended to all generated module names

### CODE REVIEW

File: lib/exprotoc/generator.ex [hunk 1]
```diff
@@ -1,14 +1,17 @@
 defmodule Exprotoc.Generator do
   @moduledoc "Given a structured AST, generate code files into an output dir."
+  
+  # prefix for module names in generated code files
+  @namespace_prefix "Proto"
```
Comment (jeremyong) [inline lib/exprotoc/generator.ex]: Did you mean to leave this here?

File: lib/compile.exprotoc.ex [hunk 2]
```diff
@@ -28,4 +29,13 @@ defmodule Mix.Tasks.Compile.Exprotoc do
       path
     end
   end
+
+  defp get_namespace do
+    namespace = Keyword.fetch Mix.project, :proto_namespace
+    if namespace == :error do
+      { :ok, nil }
```
Comment (jeremyong) [inline lib/compile.exprotoc.ex]: I actually think it makes sense for the default namespace to be the Project name.

File: lib/exprotoc/generator.ex (hunk 1 above)
Comment (johnfoconnor) [inline lib/exprotoc/generator.ex]: no i did not. Will remove

File: lib/compile.exprotoc.ex [hunk 3]
```diff
@@ -28,4 +29,22 @@ defmodule Mix.Tasks.Compile.Exprotoc do
...
+    else
+      namespace
+    end
+  end
+
+  defp to_namespace(name) do
+    name
+      |> atom_to_binary
+      |> String.split(~r{_+})
+      |> Enum.map( &(String.capitalize(&1)) )
```
Comment (jeremyong) [inline lib/compile.exprotoc.ex:47]: Are the inner parens necessary here? jw.

**Comment by jeremyong** [comment]
Is a way to introduce a root level symbol without needing all the code you needed to add to the generator? Why not just change the root symbol of the tree and then change the `search_ast` function in `ast.ex` to account for the namespace?

**Comment by jeremyong** [comment]
Any progress?

**Comment by johnfoconnor** [comment]
I've been busy with other features and haven't had time to look into refactoring this functionality to originate from the ast tree vs the generator

**Comment by johnfoconnor** [comment]
you can see some of the things i've been doing here

proper default values, transformation functions around enums, multi get and a few other small things

https://github.com/johnfoconnor/exprotoc/compare/jeremyong:master...102343e

**Comment by jeremyong** [comment]
PRs?

**Comment by johnfoconnor** [comment]
absolutely. It's just lower on the list of things to do right now

**Pull request closed without merging**